### PR TITLE
Added initcontainers to server deployment.yaml

### PR DESCRIPTION
**Pull request by smark88**
Checklist:

* [x] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA and the build is green.
* [x] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

Changes: Added initcontainers to server deployment.yaml

This gives the ability to have an init container on deploy, so if you are going to use custom plugin you can make your `deployment.yaml` similar to the one in the [documentation](https://argoproj.github.io/argo-cd/operator-manual/custom_tools/) 

```
    spec:
      # 1. Define an emptyDir volume which will hold the custom binaries
      volumes:
      - name: custom-tools
        emptyDir: {}
      # 2. Use an init container to download/copy custom binaries into the emptyDir
      initContainers:
      - name: download-tools
        image: alpine:3.8
        command: [sh, -c]
        args:
        - wget -qO- https://storage.googleapis.com/kubernetes-helm/helm-v2.12.3-linux-amd64.tar.gz | tar -xvzf - &&
          mv linux-amd64/helm /custom-tools/
        volumeMounts:
        - mountPath: /custom-tools
          name: custom-tools
      # 3. Volume mount the custom binary to the bin directory (overriding the existing version)
      containers:
      - name: argocd-repo-server
        volumeMounts:
        - mountPath: /usr/local/bin/helm
          name: custom-tools
          subPath: helm
```

### CODE REVIEW

**Comment by spencergilbert** [comment]
Based on the docs it looks like the tooling just needs to be loaded to the repoServer, which should be possible here? https://github.com/argoproj/argo-helm/blob/master/charts/argo-cd/values.yaml#L717

File: charts/argo-cd/values.yaml [hunk 1]
```diff
@@ -342,6 +342,15 @@ server:
   ##
   podLabels: {}
 
+  ## initContainer
+  initContainers: {}
```
Comment (seanson) [inline charts/argo-cd/values.yaml:346]:
```suggestion
  initContainers: []
```
This should be a list instead of a map.

**Comment by jtcarnes** [comment]
What is the status of this? I know for a number of applications for secret management additions (KSOPS, Helm secrets), the init containers would be a huge help.

**Comment by alexec** [comment]
Stale.

**Pull request closed without merging**